test/e2e/workflow/deployment_test.go

```diff
@@ -160,6 +161,41 @@ var _ = Describe("NetworkAddonsConfig", func() {
 			CheckConfigCondition(ConditionAvailable, ConditionTrue, 10*time.Minute, 30*time.Second)
 		})
 	})
+
+	Context("testing range generation after kubeMacPool reinstalation ", func() {
+		var (
+			config                                           opv1alpha1.NetworkAddonsConfigSpec
+			oldRangeStart, oldRangeEnd, rangeEnd, rangeStart string
+		)
+
+		BeforeEach(func() {
+			By("Installing KubeMacPool")
+			config = opv1alpha1.NetworkAddonsConfigSpec{KubeMacPool: &opv1alpha1.KubeMacPool{}}
+			CreateConfig(config)
+			CheckConfigCondition(ConditionAvailable, ConditionTrue, 10*time.Minute, CheckDoNotRepeat)
+			By("Should generate a non-multicast range")
+			rangeStart, rangeEnd = CheckUnicastAndValidity()
+			oldRangeStart = rangeStart
+			oldRangeEnd = rangeEnd
+
+			By("Reinstalling KubeMacPool")
+			DeleteConfig()
+			CheckComponentsRemoval(AllComponents)
+
+			config = opv1alpha1.NetworkAddonsConfigSpec{KubeMacPool: &opv1alpha1.KubeMacPool{}}
+			CreateConfig(config)
+			CheckConfigCondition(ConditionAvailable, ConditionTrue, 10*time.Minute, CheckDoNotRepeat)
+			rangeStart, rangeEnd = CheckUnicastAndValidity()
+		})
+
+		Context("after KubeMacPool has been reinstalled", func() {
+			It("should generate a new non-multicast range that is different then the old one", func() {
+				By("Checking the ranges")
```

Comment:
Now, it our `Context` is installed KubeMacPool. This test should

1. Read current range
2. Perform Reinstallation
3. Check new range